docs/dep/dep-006-new-go-sdk.md

```diff
@@ -0,0 +1,474 @@
+│   ├── writer/
+│   │   ├── sqlite/
+│   │   │   └── go.mod
+│   │   ├── remote-rpc/
+│   │   │   └── go.mod
+│   │   └── elasticsearch/
+│   │       └── go.mod
+│   └── transformer/
+│       ├── atom-enricher/
+│       │   └── go.mod
+│       ├── tagger/
+│       │   └── go.mod
+│       └── prioritiser/
+│           └── go.mod
+```
+
+The components will follow the known structure and just re-use actions or define their own
+to implement the goals of the component.
+
+One can see a scenario where an SQLite reader/writer can be used between multiple components
+to read/write Vulnerability Reports from/to.
+
+<a id="proposal-default-readers-writers"></a>
+
+### Default readers/writers
+
+We want to provide as soon as possible two reader/writer solutions for both local and saas environments.
+
+The reason is to speed up development as well as going closer to having a way to run components more easily.
+
+<a id="proposal-default-readers-writers-local"></a>
+
+#### Local
+
+For local settings, we want to use an [Embedded Database](https://en.wikipedia.org/wiki/Embedded_database) which
+is lightweight and doesn't require a standalone application or container to run.
+
+This should drastically help running components locally and driving adoption given the benefits in terms
+of development experience.
+
+Ultimately, this will be a new `Reader/Writer` part of the `Runner` contract.
+
+Our decision is to experiment with [SQLite](https://www.sqlite.org/).
+
+Some of the reasons while we decided to experiment with it are:
+
+* Self-contained: SQLite is a single library that doesn't require separate server software or configuration, making it easy to integrate directly into applications.
+* Lightweight: With a small footprint, ideal for embedded systems and applications with limited resources.
+* Zero-configuration: SQLite doesn't require a server to set up or maintain, simplifying deployment and reducing maintenance costs.
+* Cross-platform: It works on multiple operating systems, ensuring broad compatibility for embedded systems.
+* SQL compatible API: Knowing SQL is all it takes to use SQLite. There are several packages to interact with SQLite like [go-sqlite3](https://github.com/mattn/go-sqlite3).
+* Reliable: SQLite supports atomic transactions, ensuring data integrity, even in case of crashes or power failures.
+* Fast for small-scale applications: It is optimized for read-heavy workloads and provides high performance for smaller datasets, making it ideal for lightweight use cases.
+* Public domain: SQLite is free and open-source, with no licensing fees, which makes it attractive for commercial and non-commercial use.
+
+It's great in environments where a database service is not necessary.
+
+*We need to do a spike with SQLite first to finalise the decision or pivot fast.*
+
+<a id="proposal-default-readers-writers-saas"></a>
+
+#### SAAS
+
+For `SAAS` we decided to invest in a more sophisticated system to abstract reads/writes to different places.
```

Comment:
Would the SAAS include an SQLite db, to fall back on until the custom ones are configured? Thinking about what will make the SAAS easier to onboard.

Reply:
I think that to make SAAS work asap we'd have to write an integration for Postgres, to comply to how it works today - @ptzianos correct me if I'm wrong

SQLite is not designed to be used reliably by multiple applications - it's good enough for a standalone app that is "ok" losing data